src/freenas/usr/local/lib/middlewared_truenas/plugins/failover.py

```diff
@@ -169,167 +190,41 @@ def licensed(self):
-                        '/sbin/camcontrol', 'smpphylist', enclosure, '-q'
-                    ], stdout=subprocess.PIPE, stderr=subprocess.PIPE, encoding='utf8')
-                    phylist = proc.communicate()[0].strip()
-                    reg = re.search(r'ESCE A_(5[0-9A-F]{15})', encstat, re.M)
-                    if reg:
-                        addr = f'0x{(int(reg.group(1), 16) - 1):016x}'
-                        if addr in phylist:
-                            node = 'A'
-                            break
-                    reg = re.search(r'ESCE B_(5[0-9A-F]{15})', encstat, re.M)
-                    if reg:
-                        addr = f'0x{(int(reg.group(1), 16) - 1):016x}'
-                        if addr in phylist:
-                            node = 'B'
-                            break
-                else:
-                    # Identify ECHOWARP platform by one of enclosure names.
-                    reg = re.search(r'Enclosure Name: (ECStream|iX) 4024S([ps])', encstat, re.M)
-                    if reg:
-                        hardware = 'ECHOWARP'
-                        # Identify node by the last symbol of the model name
-                        if reg.group(2) == 'p':
-                            node = 'A'
-                            break
-                        elif reg.group(2) == 's':
-                            node = 'B'
-                            break
-
-        if node:
-            return hardware, node
-
-        proc = subprocess.Popen([
-            'dmidecode',
-            '-s', 'system-serial-number',
-        ], stdout=subprocess.PIPE, stderr=subprocess.PIPE, encoding='utf8')
-        serial = proc.communicate()[0].split('\n', 1)[0].strip()
-
-        license = SystemService._get_license()
-
-        if license is not None:
-            if license['system_serial'] == serial:
-                node = 'A'
-            elif license['system_serial_ha'] == serial:
-                node = 'B'
-
-        if node is None:
-            return 'MANUAL', None
-
-        if license['system_serial'] and license['system_serial_ha']:
-            mode = None
-            proc = subprocess.Popen([
-                'dmidecode',
-                '-s', 'baseboard-product-name',
-            ], stdout=subprocess.PIPE, stderr=subprocess.PIPE, encoding='utf8')
-            board = proc.communicate()[0].split('\n', 1)[0].strip()
-            # If we've gotten this far it's because we were unable to
-            # identify ourselves via enclosure device.
-            if board == 'X8DTS':
-                hardware = 'SBB'
-            elif board.startswith('X8'):
-                hardware = 'ULTIMATE'
-            else:
-                mode = 'MANUAL', None
-
-            if mode is None:
-                mode = hardware, node
-            return mode
-        else:
-            return 'MANUAL', None
+        return self.HA_MODE
 
     @accepts()
     def hardware(self):
```

Comment:
We should probably turn this into a coroutine as well

Reply:
I didn't change those to not change the API? If I change these to coroutines, I'll need to change all callers too, right?

Reply:
User API is unchanged. You need to change if they are using self and not self.middleware.call